### PR TITLE
[Paywalls V2] Adds override system via `PresentedPartial`

### PR DESCRIPTION
**Pull request by JayShortway**
Adds: 
- `PresentedPartial` and the override system
- `LocalizedTextPartial`, will add other partials in future PRs
- a bunch of tests (this is where the bulk of the lines in the diff come from)

### CODE REVIEW

**Review by tonidero — APPROVED** [review]
Left some comments but nothing really blocking. Looks good!

**Comment by joshdholtz** [comment]
@JayShortway Does this need rebased? Looks like it has the poko commit in here still? 🤔

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/TextComponent.kt [hunk 1]
```diff
@@ -61,7 +61,7 @@ internal constructor(
 @Serializable
 class PartialTextComponent
 @Suppress("LongParameterList")
-internal constructor(
+constructor(
```
[the remark this answers is not in the document]
Comment (JayShortway) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/TextComponent.kt:64]: I merged everything as closed down as possible, and I'm opening things up as I need them in RevenueCatUI. It's all still marked with `@InternalRevenueCatAPI`.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/PresentedPartial.kt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+     */
+    @get:JvmSynthetic val expanded: T?,
+)
+
+/**
+ * Converts component overrides to presented overrides.
+ */
+@Suppress("ReturnCount")
+@JvmSynthetic
+internal fun <T : PartialComponent, P : PresentedPartial<P>> ComponentOverrides<T>.toPresentedOverrides(
```
[the remark this answers is not in the document]
Comment (JayShortway) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/PresentedPartial.kt:82]: I can't say I disagree. Apparently this has a name: [curiously recurring template pattern](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern). I prefer to keep it consistent with iOS though so it's easier to reason about, and not delay shipping indeed. We can always simplify this later, as it's not public.

**Comment by JayShortway** [comment]
@joshdholtz Not that I'm aware of. It was originally based on `pw2-textcomponentview` indeed, but I've been merging `main` into it after that one was merged. All commits will be squashed when merging to `main`.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ScreenCondition.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    ;
+
+    companion object {
+        @JvmSynthetic
+        fun from(sizeClass: WindowWidthSizeClass) =
+            when (sizeClass) {
+                WindowWidthSizeClass.COMPACT -> COMPACT
+                WindowWidthSizeClass.MEDIUM -> MEDIUM
+                WindowWidthSizeClass.EXPANDED -> EXPANDED
+                else -> COMPACT
```
[the remark this answers is not in the document]
Comment (JayShortway) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ScreenCondition.kt]: That's a good point! Done here: https://github.com/RevenueCat/purchases-android/pull/1948/commits/bdae02e8e548495a02782e700cd4c197a3a990a3.

**Pull request merged**